File: BlogApp-RN/src/features/posts/domain/usecases/deletePost.ts
import { ResponseMessage } from "@/src/features/authentication/data/models/responseMessage";
import { AsyncResult } from "../../../../core/typedefs/asyncResult";
import PostRepo from "../repos/postRepo";

class DeletePost {
  constructor(private repo: PostRepo) {}
  call: (postId: string) => AsyncResult<ResponseMessage> = (postId) => {
    return this.repo.deletePost(postId);
  };
}

export default DeletePost;

File: blogapp_flutter/lib/features/posts/domain/usecases/delete_post.dart
import 'package:blogapp_flutter/core/typedefs/async_result.dart';
import 'package:blogapp_flutter/core/typedefs/usecases.dart';
import 'package:blogapp_flutter/features/authentication/data/models/response_message.dart';
import 'package:blogapp_flutter/features/posts/domain/repos/posts_repo.dart';

class DeletePost extends UsecaseWithParams<ResponseMessage, String> {
  final PostsRepo _repo;
  DeletePost(this._repo);
  @override
  AsyncResult<ResponseMessage> call(params) {
    return _repo.deletePost(params);
  }
}
